packages/ngtools/webpack/src/transformers/find_image_domains_spec.ts

```diff
@@ -0,0 +1,153 @@
+            type: NgModule,
+            args: [{
+                    declarations: [
+                        AppComponent
+                    ],
+                    imports: [
+                        BrowserModule,
+                        NgOptimizedImage
+                    ],
+                    providers: [
+                        ${provider}
+                    ],
+                    bootstrap: [AppComponent]
+                }]
+        }], null, null); })();
+    (function () { (typeof ngJitMode === "undefined" || ngJitMode) && i0.ɵɵsetNgModuleScope(AppModule, { declarations: [AppComponent], imports: [BrowserModule,
+            NgOptimizedImage] }); })();
+    `;
+}
+
+describe('@ngtools/webpack transformers', () => {
+  describe('find_image_domains', () => {
+    it('should find a domain when a built-in loader is used with a string-literal-like argument', () => {
+      // Intentionally inconsistent use of quote styles in this data structure:
+      const builtInLoaders: Array<[string, string]> = [
+        ['provideCloudflareLoader("www.cloudflareUrl.com")', 'www.cloudflareUrl.com'],
+        [
+          "provideCloudinaryLoader('https://www.cloudinaryUrl.net')",
+          'https://www.cloudinaryUrl.net',
+        ],
+        ['provideImageKitLoader("www.imageKitUrl.com")', 'www.imageKitUrl.com'],
+        ['provideImgixLoader(`www.imgixUrl.com/images/`)', 'www.imgixUrl.com/images/'],
+      ];
+      for (const loader of builtInLoaders) {
+        const input = inputTemplate(loader[0]);
+        const result = Array.from(findDomains(input));
+        expect(result.length).toBe(1);
+        expect(result[0]).toBe(loader[1]);
+      }
+    });
+
+    it('should not find a domain when a built-in loader is used with a variable', () => {
+      const input = inputTemplate(`provideCloudflareLoader(myImageCDN)`);
+      const result = Array.from(findDomains(input));
+      expect(result.length).toBe(0);
+    });
+
+    it('should not find a domain when a built-in loader is used with an expression', () => {
+      const input = inputTemplate(
+        `provideCloudflareLoader("https://www." + (dev ? "dev." : "") + "cloudinaryUrl.net")`,
+      );
+      const result = Array.from(findDomains(input));
+      expect(result.length).toBe(0);
+    });
+
+    it('should not find a domain when a built-in loader is used with an template literal', () => {
+      const input = inputTemplate(
+        'provideCloudflareLoader(`https://www.${dev ? "dev." : ""}cloudinaryUrl.net`)',
+      );
+      const result = Array.from(findDomains(input));
+      expect(result.length).toBe(0);
+    });
+
+    it('should not find a domain in a function that is not a built-in loader', () => {
+      const input = inputTemplate('provideNotARealLoader("https://www.foo.com")');
+      const result = Array.from(findDomains(input));
+      expect(result.length).toBe(0);
+    });
+
```

Comment:
It would be good to have a test where there are multiple providers (maybe 3 with the image one in the middle?)

Reply:
Good call. I added one at the end with multiple providers/provider types